authors/apps/authentication/tests/test_login.py

```diff
@@ -0,0 +1,48 @@
+""" module to test login. """
+from django.urls import reverse
+from rest_framework import status
+from rest_framework.test import APITestCase
+
+from authors.apps.authentication.models import User
+
+
+class AuthenticationTests(APITestCase):
+    def setUp(self):
+        """ Setup data for the tests """
+        self.user1 = {
+            "username": "createduser1",
+            "email": "createduser1@user.user",
+            "password": "user123user"
+        }
+        self.reg_url = reverse('authentication:register')
+        self.login_url = reverse('authentication:login')
+        reg = self.client.post(self.reg_url, self.user1, format='json')
+
+    def test_successful_login_user(self):
+        """ Test that a user successfully logs in """
+        response = self.client.post(self.login_url, self.user1, format='json')
+        self.assertEqual(login_response.status_code, status.HTTP_201_CREATED)
```

Comment:
What happens when a user is successfully logged in? As in, are they assigned an *authorization token* or a *session* variable of some sort? If so, then I suggest you include an assertion test for the authorization variable. **WHY?** Because a redirection can happen and you still get *HTTP_201_CREATED* if the server hit returns this code.

Reply:
I have noted that feedback, We shall implement the testing when we have a clear know-how of our authentication method.